src/identities/mnemonic.ts

```diff
@@ -355,8 +355,8 @@ export default class Mnemonic extends Identity implements IdentityInterface {
       const restoredAddresses = await this.restoreAddresses();
       restoredAddresses.forEach(addr => this.persistAddressToCache(addr));
       return true;
-    } catch (e) {
-      return false;
+    } catch (error) {
+      throw new Error(`Mnemonic restoration error: ${error}`);
```

Comment:
Just lift the error up. We will craft the error message in the upper layer where we catch it